pyproject.toml

```diff
@@ -56,7 +56,7 @@ ipython = "^8.10.0"
 qulacs = { version = "^0.6.4", markers = "(sys_platform == 'darwin' and python_version > '3.9') or sys_platform != 'darwin'" }
 seaborn = "^0.13.2"
 ipykernel = "^6.29.4"
-qibojit = { git = "https://github.com/qiboteam/qibojit.git" }
+qibojit = { git = "https://github.com/qiboteam/qibojit.git", branch="numba_threads" }
```

Comment:
Since here we do not have the same problem of tests (there isn't anything circular, since the changes in Qibojit do not depend on changes here), I'd suggest first merging on the other side, and switch back to the default branch before merging this to `master`.